### DISCUSSION TITLE
Allow relational interfaces with any layout

### DISCUSSION THREAD

**Discussion by M-GRV** [directus/directus#21021] [Feature Requests]
### Summary

When showing relational data within an item, you are currently only limited to table/list views. We should be able to select any of the available layouts to show the data how we want.

### Motivation

Allows more control over how we see and use our data in the Data Studio. Makes it easier to view relational data by logo for example which can be much easier to find vs looking at a wall of text. (use case varies)

### Detailed Design

The design is to mimic the layout tool that is used when viewing a collection in the Data Studio currently.

Old (limited to 2 views): 
![image](https://github.com/directus/directus/assets/130182174/08121849-7427-4c72-8abc-c2a1b4a65bf2)
New (includes all available views):
![image](https://github.com/directus/directus/assets/130182174/cf71d706-ec27-4374-899c-17d74660b349)

### Must Have:
- A - Allow more layout choices
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E - Any API changes

### Drawbacks

No drawbacks that I can conclude from this request.

### Alternatives

The alternative is a "no build". Where we just keep only 2 options for the layout for viewing relational data. The only impact is that we remain limited on what layouts are available for relational data.

### Adoption Strategy

Should be no breaking changes. Just offers more ways to digest your data.

**Comment by BlackDahlia313** [comment]
Hey I know that guy!

**Comment by br41nslug** [comment]
While there is some overlap in the naming the relational interfaces are not the same as the dashboard layouts so it's less of an "allow more layouts" (implying they already exist) and more "build new relational layouts". 

Please do elaborate on your requirements like what specific layout would you like to see implemented with what features/drawbacks/must-haves and so on.

**Reply by M-GRV** [reply]
Working on this now.

**Comment by alepietrobon** [comment]
Would be great to also have a display option as dropdown.
![image](https://github.com/directus/directus/assets/8207502/07b0ca55-d9bb-4f07-8efe-961344202612)
